### ISSUE TITLE
feat(radtran): optimal-estimation iterative retrieval loop

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#96]
## Problem / Request
The only inverse paths in the RTM stack are the matched filter (`radtran/matched_filter.py`, `matched_filter/`) and a one-shot linear `gaussx_solve`. There is no Rodgers optimal-estimation loop that iterates the nonlinear forward (`radtran.forward_nonlinear` through `assimilation.RadianceObservationModel`) to a state posterior with its covariance and averaging kernel — the design's Step 2 (`radtran.retrieval`) is ☐, and its note "clarify scope vs. `gaussx_solve.py`" is resolved here as: `gaussx_solve` stays the linear matched-filter path; `retrieval` owns the iterated nonlinear loop.

## User Story
> As a user with band radiances from EMIT, I want an OE retrieval that returns `(x̂, Ŝ, A)` for the methane column (and surface/aerosol nuisance terms), so Tier IV can fuse a proper posterior rather than a matched-filter score.

## Proposed API
```python
# src/plumax/radtran/retrieval.py
@dataclass(frozen=True)
class OEResult:
    x_hat: jax.Array; S_hat: lineax.AbstractLinearOperator; averaging_kernel: jax.Array
    chi2: float; dofs: float; n_iter: int; converged: bool

def optimal_estimation(
    y: jax.Array, *, model: RadianceObservationModel, x_a: jax.Array, S_a: lineax.AbstractLinearOperator,
    S_e: lineax.AbstractLinearOperator, method: Literal["gauss_newton", "psas"] = "gauss_newton", max_iter: int = 10, tol: float = 1e-3,
) -> OEResult: ...
```

## Implementation Steps
- [ ] Add `src/plumax/radtran/retrieval.py`; outer loop = `assimilation.solve.run_gauss_newton` / `run_dual_psas` with `S_a` as `B` and `S_e` as `R`
- [ ] `S_hat = (Kᵀ S_e⁻¹ K + S_a⁻¹)⁻¹` and `A = S_hat Kᵀ S_e⁻¹ K` via `gaussx` (low-rank when `dim(y)` large)
- [ ] Convergence on `dᵢ² = δxᵀ Ŝ⁻¹ δx < tol · n` (Rodgers §5.6.3), surfaced like `FourDVarResult.converged`
- [ ] Export from `radtran/__init__.py` lazily; API page entry

## Definition of Done
- [ ] Synthetic-truth recovery: on HAPI-generated radiances the retrieved column sits inside `Ŝ` ~68 % of the time over ≥ 200 trials
- [ ] Linear limit: one Gauss-Newton step equals `gaussx_solve.matched_filter_pixel_op` for a linear target
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/radtran/test_retrieval.py::test_coverage_68pct` (`slow`)
- [ ] `tests/radtran/test_retrieval.py::test_linear_limit_matches_gaussx_solve`

## Relationships
- Parent (theme epic): #73
- Blocks: quality flags + information-content diagnostics, Tier IV nonlinear end-to-end inversion
- Related: #43 (x64 in gaussx_solve, closed)
